### PR TITLE
More benchmark fixes

### PR DESCRIPTION
**Pull request by richafrank**
Fixes https://github.com/quantopian/zipline/issues/2761 and adds the docstring missing in https://github.com/quantopian/zipline/issues/2627#issuecomment-675127746.

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -73,9 +73,20 @@ def test_example(self, example_name):
...
                 'ZIPLINE_ROOT': self.tmpdir.getpath('example_data/root'),
             },
-            benchmark_returns=self.benchmark_returns,
+            benchmark_returns=benchmark_returns,
         )
-        expected_perf = self.expected_perf[example_name]
+        if benchmark_returns is not None:
+            expected_perf = self.expected_perf[example_name]
+        else:
+            # With no benchmark, expect zero results for these metrics:
```
Comment (llllllllll) [inline tests/test_examples.py:81]: should we precompute this in the init like `self.no_benchmark_expected_perf`?

File: zipline/utils/run_algo.py [hunk 2]
```diff
@@ -513,7 +515,8 @@ def resolve(self, asset_finder, start_date, end_date):
                 benchmark_returns = None
             except SymbolNotFound:
                 raise _RunAlgoError(
-                    "Symbol %s as a benchmark not found in this bundle."
+                    "Symbol %r as a benchmark not found in this bundle."
+                    % self.benchmark_symbol
```
Comment (llllllllll) [inline zipline/utils/run_algo.py:519]: +1

**Comment by llllllllll** [comment]
lgtm

File: tests/test_examples.py (hunk 1 above)
Comment (richafrank) [reply tests/test_examples.py]: Good idea. Done.

**Comment by RichardDale** [comment]
Given this is such a common-use case I would suggest this probably warrants a v1.41 release?  It's certainly been a common topic from our user base.

**Comment by richafrank** [comment]
> Given this is such a common-use case I would suggest this probably warrants a v1.41 release? It's certainly been a common topic from our user base.

👍 I was thinking the same thing.

**Pull request merged**